### PR TITLE
WIP: PBuffer support and Fix for #12

### PR DESCRIPTION
**Pull request by devshgraphicsprogramming**
@Crisspl feel free to add more to the description

### CODE REVIEW

File: EGL/include/EGL/eglctxinternals.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
+#ifndef EGL_CTXINTERNALS_H_
+#define EGL_CTXINTERNALS_H_
+
+#if defined(_WIN32) || defined(__VC32__) && !defined(__CYGWIN__) && !defined(__SCITECH_SNAP__) /* Win32 and WinCE */
```
Comment (devshgraphicsprogramming) [inline EGL/include/EGL/eglctxinternals.h:4]: dont worry about WinCE, its dead

File: EGL/src/egl_x11.cpp [hunk 2]
```diff
@@ -43,11 +44,36 @@ __PFN_glFinish glFinish_PTR = NULL;
 #define glXSwapIntervalEXT(...) glXSwapIntervalEXT_PTR(__VA_ARGS__)
 #define glXCreateContextAttribsARB(...) \
     glXCreateContextAttribsARB_PTR(__VA_ARGS__)
-#endif 
+#endif
+
+void* libx11 = NULL;
+void* libgl = NULL;
+//X
```
Comment (devshgraphicsprogramming) [inline EGL/src/egl_x11.cpp:51]: any stuff used from `xrandr`, `xxf86vm` ?

Baiscally does `ldd` come back clean?

**Comment by McNopper** [comment]
Please ping me, when you think it is ready to be merged.

**Comment by Crisspl** [comment]
> Please ping me, when you think it is ready to be merged.

It's ready to merge as it is now (it's working on both Linux and Windows), but I'm continously adding new stuff when there is a need for our Nabla engine

**Comment by Crisspl** [comment]
Btw, besides making it compliant with the EGL spec, this PR also adds support for:
* creating GLES3 contexts, 
* safe concurrent (multi-threaded) usage
* pbuffer support
* sRGB surfaces support
* 100% dynamic loading of all X11, glX and WGL functions being used
* a fix so that `eglCreateContext` actually returns `EGL_NO_CONTEXT` if requested version is greater than maximum supported (before the fix, `eglCreateContext` would succeed, but `eglMakeCurrent` would fail)
* more less important features (see commits)

File: EGL/src/egl_x11.cpp (hunk 2 above)
Comment (Crisspl) [reply EGL/src/egl_x11.cpp:51]: > any stuff used from xrandr, xxf86vm ?

nah, none

> Baiscally does ldd come back clean?

yep

**Comment by devshgraphicsprogramming** [comment]
> 
> 
> > Please ping me, when you think it is ready to be merged.
> 
> It's ready to merge as it is now (it's working on both Linux and Windows), but I'm continously adding new stuff when there is a need for our Nabla engine

I think its worth mentioning that we tried to test on Wayland, but ran into difficulties 
https://github.com/Devsh-Graphics-Programming/Nabla/issues/96

**Comment by Crisspl** [comment]
> I think its worth mentioning that we tried to test on Wayland, but ran into difficulties

not really related to this project

**Comment by devshgraphicsprogramming** [comment]
> 
> 
> > I think its worth mentioning that we tried to test on Wayland, but ran into difficulties
> 
> not really related to this project

how else can we test that this EGL<->system EGL passthrough path works?

**Pull request merged**